chocolatey/choco
# (#3489) Return early when folder doesn't exist

## Description Of Changes

This commit addresses the issue by first checking to see if the package folder exists.  If it doesn't, then it doesn't attempt to create the .chocolateyPending file, and simply returns early.  This then allows the remaining operations, i.e. returning the package folder from the lib-bkp folder, back to the lib folder.

## Motivation and Context

The MarkPackagePending method is used to create a .chocolateyPending file in the package folder to indicate that there is a pending operation that needs to be completed.

However, during an upgrade operation, the package folder is moved to the lib-bkp folder, and as a result, doesn't exist anymore in the lib folder.

If during the upgrade operation, something fails, for example, an exception when trying to download the nupkg file, a process of returning the package folder to the correct place begins.  The first thing that is attempted is to create the .chocolateyPending file in the package folder.  However, as mentioned, this folder no longer exists in the expected location.  As a result, it is possible for an exception to be thrown, as the file cannot be created in the package folder.

This problem can lead to another related problem in that the package is now "lost" from lib folder, and will not show up when doing for example choco list.

Imagine you were attempting the following command, and an exception happened during the downloading of the nupkg:

choco upgrade audacity

Chocolatey CLI would start the upgrade operation, and then it would fail downloading the package.  It would attempt to create the .chocolateyPending file, and another exception would be thrown.  The package folder is not returned to the lib folder, and now the package is "lost".  If you attempt to run choco upgrade audacity again, the operation would succeed, as it would see that the package is not installed and it would be installed, however, if this happened during the process of running choco upgrade all, then the underlying audacity application would not be upgraded, as Chocolatey CLI is no longer managing the package.

## Testing

In order to test this, you have to mimic an exception being thrown during the download of the nupkg package.  The simplest way to do that is to add the following line of code after line 1619 of the NugetService.cs file.

```
throw new Exception("Mimic failure of package download");
```

With this change in place, the code should look like the following:

<img width="1504" alt="image" src="https://github.com/user-attachments/assets/e465cf41-5c72-4d77-9b02-ac29eefe45b8">

1. Run `choco install audacity --version 3.6.0 -n`
2. The audacity package should be installed without any issues
3. Run `choco list` to verify that the package is installed and listed as version 3.6.0
4. Run `choco upgrade audacity -n`
5. This operation will fail, and it will state that the package wasn't upgrade
6. Run `choco list` to verify that 3.6.0 package of audacity is still there

### Operating Systems Testing

- Windows 10

## Change Types Made


## Change Checklist


## Related Issue

Fixes #3489
